File: library/src/androidTest/mock/com/mobilesolutionworks/android/app/test/configuration/TestWorksController.java
package com.mobilesolutionworks.android.app.test.configuration;

import android.content.res.Configuration;

import com.mobilesolutionworks.android.app.controller.WorksController;

/**
 * Created by yunarta on 12/3/17.
 */

public class TestWorksController extends WorksController {

    public int configChangesCount = 0;

    @Override
    public void onConfigurationChanged(Configuration config) {
        super.onConfigurationChanged(config);
        configChangesCount++;
    }
}

File: test-app/src/main/kotlin/com/mobilesolutionworks/android/controller/test/basic/TestWorksController.kt
package com.mobilesolutionworks.android.controller.test.basic

import android.content.res.Configuration

import com.mobilesolutionworks.android.app.controller.WorksController
import com.mobilesolutionworks.android.app.controller.WorksControllerManager

/**
 * Created by yunarta on 12/3/17.
 */

class TestWorksController(manager: WorksControllerManager) : WorksController(manager) {

    var configChangesCount = 0

    override fun onConfigurationChanged(config: Configuration) {
        super.onConfigurationChanged(config)
        configChangesCount++
    }
}
